Src/GoogleApis.Auth.DotNet4/OAuth2/ServiceAccountCredential.cs

```diff
@@ -72,6 +76,20 @@ public Initializer(string id, string tokenServerUrl) : base(tokenServerUrl)
                 Scopes = new List<string>();
             }
 
+            /// <summary>Constructs a new initializer using the given id and the private key in the ClientCredentialParameters.</summary>
+            public Initializer(ClientCredentialParameters clientCredentialParameters)

```

Comment:
to initialize from X509Certificate2  there's a FromCertificate factory method. I think we should honor this design and add a FromJsonKey() factory method instead of adding a constructor.